video/src/ai/eto/rikai/sql/spark/datasources/VideoDataSourceV2.scala

```diff
@@ -50,5 +50,6 @@ class VideoDataSourceV2 extends FileDataSourceV2 {
     )
   }
 
-  override def fallbackFileFormat: Class[_ <: FileFormat] = ???
+  override def fallbackFileFormat: Class[_ <: FileFormat] =
```

Comment:
From the document I saw
```
   * This is a solution for the following cases:
   * 1. File datasource V2 implementations cause regression. Users can disable the problematic data
   *    source via SQL configuration and fall back to FileFormat.
   * 2. Catalog support is required, which is still under development for data source V2.
```
Which one is the use case here?

Reply:
If there were some links in the doc shows how spark will use this, it would be great.

Reply:
comments added

VideoFileFormat is for the following syntax:

```
select * from video.`/path/to/video.mp4`
```

Reply:
Apache Spark is migrating from old XXXFileFormat to DataSourceAPIv2. No docs found on spark.apache.org. Currently, one should read the source code of Apache Spark.